### PR TITLE
fix: delegation bug

### PR DESCRIPTION
**Pull request by sakulstra**

### CODE REVIEW

File: tests/BaseTest.sol [hunk 1]
```diff
@@ -83,7 +83,10 @@ contract BaseTest is Test {
 
   function _stake(uint256 amount, address user) internal {
     deal(address(STAKE_CONTRACT.STAKED_TOKEN()), user, amount);
-    STAKE_CONTRACT.STAKED_TOKEN().approve(address(STAKE_CONTRACT), amount);
+    STAKE_CONTRACT.STAKED_TOKEN().approve(
+      address(STAKE_CONTRACT),
+      type(uint256).max
```
Comment (eboadom) [inline tests/BaseTest.sol:88]: Why this change to max here @sakulstra ?

**Review by eboadom — DISMISSED** [review]

**Review by kyzia551 — DISMISSED** [review]
